### PR TITLE
any-nix-shell: 1.2.1 -> 2.0.0

### PR DESCRIPTION
**Pull request by thiagokokada**
## Description of changes

~~The [current](https://github.com/haslersn/any-nix-shell/releases/tag/v1.2.1) stable version is more than 3 years old, and is missing some important fixes, especially [this](https://github.com/haslersn/any-nix-shell/pull/13) one that adds support to `nix run` and fixes `nix shell`.~~

~~Also backport PR [34](https://github.com/haslersn/any-nix-shell/pull/34), that adds support to `nix develop`.~~

New stable release: https://github.com/haslersn/any-nix-shell/releases/tag/v2.0.0

Misc changes include moving to `pkgs/by-name`, formatting it accordingly to RFC166, and adding missing dependencies to the package (e.g.: GNU sed).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/an/any-nix-shell/package.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        which
+      ]
+    }
+  '';
+
+  meta = with lib; {
+    description = "fish, xonsh and zsh support for nix-shell";
+    license = licenses.mit;
+    homepage = "https://github.com/haslersn/any-nix-shell";
+    maintainers = with maintainers; [ haslersn ];
```
Comment (AndersonTorres) [inline pkgs/by-name/an/any-nix-shell/package.nix:51]: Is theey active??

Comment (thiagokokada) [reply pkgs/by-name/an/any-nix-shell/package.nix]: Not sure, but I don't think I should remove it in this PR.

Maybe a dedicated PR to remove them.

Comment (haslersn) [reply pkgs/by-name/an/any-nix-shell/package.nix]: Yes, I'm active

**Comment by thiagokokada** [comment]
> You could have just requested a new release at the https://github.com/haslersn/any-nix-shell repo.

If this is possible, please. I was assuming since there is no activity in the repository that it was dead.

**Comment by thiagokokada** [comment]
Converting this PR to draft until we have a new release since @haslersn asked.

**Review by AndersonTorres — APPROVED** [review]
LGTM

**Pull request merged**